tracing-attributes/src/attr.rs

```diff
@@ -375,9 +377,47 @@ impl Parse for Level {
         let _ = input.parse::<Token![=]>()?;
         let lookahead = input.lookahead1();
         if lookahead.peek(LitStr) {
-            Ok(Self::Str(input.parse()?))
+            let str: LitStr = input.parse()?;
+            if str.value().eq_ignore_ascii_case("trace") {
+                Ok(Level::Trace)
+            } else if str.value().eq_ignore_ascii_case("debug") {
+                Ok(Level::Debug)
+            } else if str.value().eq_ignore_ascii_case("info") {
+                Ok(Level::Info)
+            } else if str.value().eq_ignore_ascii_case("warn") {
+                Ok(Level::Warn)
+            } else if str.value().eq_ignore_ascii_case("error") {
+                Ok(Level::Error)
+            } else {
+                Err(input.error(
+                    "unknown verbosity level, expected one of \"trace\", \
+                     \"debug\", \"info\", \"warn\", or \"error\", or a number 1-5",
+                ))
+            }
         } else if lookahead.peek(LitInt) {
-            Ok(Self::Int(input.parse()?))
+            fn is_level(lit: &LitInt, expected: u64) -> bool {
+                match lit.base10_parse::<u64>() {
+                    Ok(value) => value == expected,
+                    Err(_) => false,
+                }
+            }
+            let int: LitInt = input.parse()?;
+            if is_level(&int, 1) {
+                Ok(Level::Trace)
+            } else if is_level(&int, 2) {
+                Ok(Level::Debug)
+            } else if is_level(&int, 3) {
+                Ok(Level::Info)
+            } else if is_level(&int, 4) {
+                Ok(Level::Warn)
+            } else if is_level(&int, 5) {
+                Ok(Level::Error)
+            } else {
+                Err(input.error(
+                    "unknown verbosity level, expected one of \"trace\", \
+                     \"debug\", \"info\", \"warn\", or \"error\", or a number 1-5",
+                ))
+            }
```

Comment:
take it or leave it: could be a little more concise as a `match`?
```suggestion
            match int {
                int if is_level(&int, 1) => Ok(Level::Trace),
                int if is_level(&int, 2) => Ok(Level::Debug),
                int if is_level(&int, 3) => Ok(Level::Info),
                int if is_level(&int, 4) => Ok(Level::Warn),
                int if is_level(&int, 5) => Ok(Level::Error),
                _ => Err(input.error(
                    "unknown verbosity level, expected one of \"trace\", \
                    \"debug\", \"info\", \"warn\", or \"error\", or a number 1-5",
                )),
            }
```

Reply:
Done.